src/js/jsx/PanelSet.jsx

```diff
@@ -170,6 +170,12 @@ define(function (require, exports, module) {
                 prevProps.singleColumnModeEnabled !== this.props.singleColumnModeEnabled) {
                 this._updatePanelSizes();
             }
+            
+            // Turn on the ready state to start fade-in transition for panels. To ensure CSS transition, the 
+            // state must be set after the panel components are flushed to the DOM.
+            if (!this.state.ready && this._hasActiveDocument()) {
```

Comment:
Would you mind adding to this comment, to help make it obvious later, that this will only happen once?